### PR TITLE
feat(compiler): add floor division support

### PR DESCRIPTION
**Pull request by flyingImer**
Get `floor division` implemented: https://github.com/winglang/wing/issues/470

*By submitting this pull request, I confirm that my contribution is made under the terms of the [Monada Contribution License](https://docs.winglang.io/terms-and-policies/contribution-license.html)*.

### CODE REVIEW

**Comment by flyingImer** [comment]
still need to clean up the jsify part. Ideas welcome!

File: libs/wingc/src/jsify.rs [hunk 1]
```diff
@@ -371,6 +371,13 @@ impl<'a> JSifier<'a> {
 					BinaryOperator::Sub => "-",
 					BinaryOperator::Mul => "*",
 					BinaryOperator::Div => "/",
+					BinaryOperator::FloorDiv => {
+						return format!(
+							"Math.floor({} / {})",
```
Comment (staycoolcall911) [inline libs/wingc/src/jsify.rs:376]: hmmm... no special ideas how to simplify that @flyingImer. @yoav-steinberg - do you have any idea?

Comment (flyingImer) [reply libs/wingc/src/jsify.rs:376]: LMK

Will keep it in back of my mind as digging more into the art part of coding rust

File: examples/tests/valid/expressions_binary_operators.w [hunk 2]
```diff
@@ -6,3 +6,5 @@ let xyz = ((y * y) / (x * x)) * z;
 let xf = 1.0;
 let yf = (-20.22) * xf;
 let zf = (xf + (yf) - -0.01);
+let xyzf = 501 \ (99 + 1);
+assert(xyzf == 5);
```
Comment (yoav-steinberg) [inline examples/tests/valid/expressions_binary_operators.w:10]: I'm assuming we're aiming for integer division here. So lets also add a test for negative numbers which will show that using `Math.floor(..)` is a bug here.
```suggestion
let xyzf = 501 \ (99 + 1);
assert(xyzf == 5);
let xyznf = -501 \ (99 + 1);
assert(xyznf == -5);
```

File: libs/wingc/src/jsify.rs (hunk 1 above)
Comment (yoav-steinberg) [inline libs/wingc/src/jsify.rs:376]: to deal with negatives
```suggestion
							"Math.trunc({} / {})",
```

Comment (yoav-steinberg) [reply libs/wingc/src/jsify.rs:376]: Maybe do a `let js_left = self.jsify_expression(left, phase);` and `let js_right = self.jsify_expression(right, phase);` before entering this `match`, to de-dup the call to `jsify_expression`. But other than that this seems fine.

**Review by yoav-steinberg — CHANGES_REQUESTED** [review]
nice :rocket:

File: libs/wingc/src/jsify.rs (hunk 1 above)
Comment (flyingImer) [reply libs/wingc/src/jsify.rs:376]: TIL

File: examples/tests/valid/expressions_binary_operators.w [hunk 3]
```diff
@@ -6,5 +6,13 @@ let xyz = ((y * y) / (x * x)) * z;
 let xf = 1.0;
 let yf = (-20.22) * xf;
 let zf = (xf + (yf) - -0.01);
+let fxzy = 5 ^ (1^5);
+assert(fxzy == 5);
```
Comment (yoav-steinberg) [inline examples/tests/valid/expressions_binary_operators.w:10]: According to the spec this is the "exponent" operator. I'm not sure what this means. It's probably supposed to be a **power** operator. But in either case what you added here acts like the javascript `^` operator which is a "bitwise xor" operator. So to clarify:
```
// JavaScript
7^3 = 4 // Bitwise XOR: binary 111 xor binary 011 = binary 100 = 4
// Power  - this is what I think we want with the ^ operator
7^3 = Math.pow(7,3) = 343
// Exponent - x^y - this is unclear. It may mean the exponent of x but then this is a unary operator and there's no need for y
Math.exp(7)
```
@staycoolcall911 I suggest we either delete `^` from the spec for now, or fix the spec to say it's a power operator (and not exponent).

**Review by yoav-steinberg — CHANGES_REQUESTED** [review]
This seems fine, see my comment re "exponent". I suggest we either clarify this with @staycoolcall911 or just remove it from the PR so we can get it merged quickly.

File: examples/tests/valid/expressions_binary_operators.w (hunk 3 above)
Comment (flyingImer) [reply examples/tests/valid/expressions_binary_operators.w:10]: @yoav-steinberg to be clear, the "exponent" operator (https://github.com/winglang/wing/pull/1654) was jsified as `**`, which is function-wise equivalent to `Math.pow`. Tho the meaning of syntax `^` in the spec is a bit different from the javascript. 

I think it can be a separate PR if @staycoolcall911 thinks "exponent" the spec and the implementation should be removed. WDYT

Comment (staycoolcall911) [reply examples/tests/valid/expressions_binary_operators.w:10]: Good catch @yoav-steinberg.

I suggest:
1. Since this PR is not related to "exponent", I think @yoav-steinberg can approve it as is ("exponent" was actually added in a previous PR #1654 which is already merged in).
2. @flyingImer - could you please add another PR to fix the name of `exponent` to `power` in the spec? 
I'd say please also change the operator from `^` to `**`, because `^` is confusing. Thanks!

Happy to hear if there are other opinions/suggestions.

Comment (yoav-steinberg) [reply examples/tests/valid/expressions_binary_operators.w:10]: And if this PR adds a test for power (exponen) lets make it a test that really shows this is power:
```suggestion
let fxzy = 5 ^ (2^3);
assert(fxzy == 390625);
```

Comment (flyingImer) [reply examples/tests/valid/expressions_binary_operators.w]: @yoav-steinberg will make that happen in the following PR

**Review by yoav-steinberg — APPROVED** [review]
:+1:

File: examples/tests/valid/expressions_binary_operators.w (hunk 3 above)
Comment (staycoolcall911) [reply examples/tests/valid/expressions_binary_operators.w]: Resolved the conversation and @yoav-steinberg approved your PR. Thank you for your patience and for following up @flyingImer 🙏
